### ISSUE TITLE
viewcolumns and left-aligned

### ISSUE THREAD

**Issue by chelovechishko** [vifm/vifm#383]
I want to get an extension column on the left and (after a space) a column with file name. How can i do this?

I can't forcibly insert the space to the end of the first column, nor at the start of the second column.

Also, backticks eat spaces, so i have to use underscore ('_') instead of space inside the extension column.

Example files: "`Dear g-d, look at the clock.jpeg`" and "`Great_Science_Textbooks.rscollection`"
so, i want it to be something like: 
`jpeg_Dear g-d, look at the clock`
and
`rsco_Great_Science_Textbooks`
or
`rsc…_Great_Science_Textbooks`
and very good to see (mandatory space before file name and extension aligned to the right)
`_png_For the glory of`

set viewcolumns=5{fileext},-70.1{root}...
`_jpegDear g-d, look at the clock`
`rscolGreat_Science_Textbooks`

set viewcolumns=5.4{fileext}..,-70%{root}...
`_jpegDear g-d, look at the clock`
`_…ionGreat_Science_Textbooks`

set viewcolumns=-5.1{fileext},-70{root}...
`jpeg_Dear g-d, look at the clock`
`rscolGreat_Science_Textbooks`

set viewcolumns=-5.1{fileext}..,-70{root}...
`…____Dear g-d, look at the clock`
`…____Great_Science_Textbooks`

**Comment by xaizek** [comment]
If you're not using `'sortgroups'`, this seems to be quite close:
```vim
set viewcolumns=5{fileext}.,1{groups},-70{root}...
```
There was a [question on Q&A site](https://q2a.vifm.info/343/mc-like-look?show=344#a344) about inserting pseudo-graphic symbols between columns:
```vim
set viewcolumns=*{name}..,│,12{size},│,6{}.
```
I guess this is what's needed here too.

**Comment by chelovechishko** [comment]
> If you're not using `'sortgroups'`

Didn't know about that :)

```shell
set viewcolumns=5{fileext}..,1{groups},-70{root}...
```

`_jpeg Dear g-d, look at the clock`
`…tion Great_Science_Textbooks`

> ```shell 
> set viewcolumns=5{fileext}.,1{groups},-70{root}...
> ```

`_jpeg Dear g-d, look at the clock`
`ction Great_Science_Textbooks`

```shell
set viewcolumns=5{fileext},1{groups},-70{root}...
```
`_jpeg Dear g-d, look at the clock`
`rscollGreat_Science_Textbooks`
 
___
> ```shell
> set viewcolumns=*{name}..,│,12{size},│,6{}.
> ```

Oh, didn't think about that.
...
```shell
set viewcolumns=5{fileext},|,-70{root}...
```
Gives message `Trailing characters` and only the fileext column.

```shell
set viewcolumns=*{name}..,│,12{size},│,6{}.
```
Gives `Invalid format of 'viewcolumns' option`
`Invalid argument for :set command`
___
TL;DR
And that it is (and as simple as it):
```shell
set viewcolumns=5.1{fileext},-1{type},-70{root}...
```
`{nitems}` seems also goood as far as i can tell.

**Comment by xaizek** [comment]
> Gives message `Trailing characters` and only the fileext column.

It's not there, it was requested. I just noted that it would solve this issue, because instead of `|` there could be a space.

> `{nitems}` seems also goood as far as i can tell.

It incurs some performance overhead, `{type}` doesn't.

**Comment by chelovechishko** [comment]
> It's not there

Ahh, now i see what you meant.
